Card/src/main/java/com/braintreepayments/api/card/BaseCard.kt

```diff
@@ -0,0 +1,116 @@
+package com.braintreepayments.api.card
+
+import androidx.annotation.RestrictTo
+import com.braintreepayments.api.core.PaymentMethod
+import org.json.JSONException
+import org.json.JSONObject
+
+/**
+ * Base class used to build various types of cards
+ */
+abstract class BaseCard(
+    open val cardholderName: String? = null,
+    open val number: String? = null,
+    open val company: String? = null,
+    open val countryCode: String? = null,
+    open val cvv: String? = null,
+    open var expirationMonth: String? = null,
+    open var expirationYear: String? = null,
+    open val extendedAddress: String? = null,
+    open val firstName: String? = null,
+    open val lastName: String? = null,
+    open val locality: String? = null,
+    open val postalCode: String? = null,
+    open val region: String? = null,
+    open val streetAddress: String? = null,
+    override var integration: String? = null,
+    override var source: String? = null,
+    override var sessionId: String? = null,
+    override var apiPath: String = "credit_cards"
+) : PaymentMethod(integration, source, sessionId, apiPath) {
+
+    companion object {
+        const val BILLING_ADDRESS_KEY: String = "billingAddress"
+        const val CARDHOLDER_NAME_KEY: String = "cardholderName"
+        const val COMPANY_KEY: String = "company"
+        const val COUNTRY_CODE_ALPHA3_KEY: String = "countryCodeAlpha3"
+        const val COUNTRY_CODE_KEY: String = "countryCode"
+        const val CREDIT_CARD_KEY: String = "creditCard"
+        const val CVV_KEY: String = "cvv"
+        const val EXPIRATION_MONTH_KEY: String = "expirationMonth"
+        const val EXPIRATION_YEAR_KEY: String = "expirationYear"
+        const val EXTENDED_ADDRESS_KEY: String = "extendedAddress"
+        const val FIRST_NAME_KEY: String = "firstName"
+        const val LAST_NAME_KEY: String = "lastName"
+        const val LOCALITY_KEY: String = "locality"
+        const val NUMBER_KEY: String = "number"
+        const val POSTAL_CODE_KEY: String = "postalCode"
+        const val REGION_KEY: String = "region"
+        const val STREET_ADDRESS_KEY: String = "streetAddress"
+    }
+
+    var expirationDate: String?
+        get() {
+            if (expirationMonth?.isNotEmpty() == true && expirationYear?.isNotEmpty() == null) {
```

Comment:
```suggestion
            if (expirationMonth?.isNotEmpty() == true && expirationYear?.isNotEmpty() == true) {
```

Should we check for true in both cases?

Reply:
One is for Month and one is for Year.

Reply:
Sorry I should have made it more clear. For the month condition we're checking if `expirationMonth?.isNotEmpty()` is true. But for the year condition we're checking if `expirationYear?.isNotEmpty()` is null. Should these conditions align and check fo true instead of null?

Reply:
Yeah for some reason the second boolean equality check is checking against `null` and it should be `true`. Might be an existing bug that was never detected on the Java side.

Reply:
Actually, I'd be up for removing this `expirationDate` convenience accessor. The `MM/YY` format of the string it produces is somewhat opinionated.

One of our goals should be to make the SDK less opinionated in general, and removing this property is in line with that goal.

Reply:
@sshropshire - that's a great call! it should be up to the merchant on how they wish to display their formatted date. I'm in favor of removing this accessor.